### PR TITLE
[master] enable bip86 signing, cleanup and add tests

### PR DESCRIPTION
**Pull request by gregdhill**
Same as https://github.com/bob-collective/btcsnap/pull/8 but cherry-picked to master

Closes https://github.com/bob-collective/btcsnap/issues/4

### CODE REVIEW

**Review by tomjeatt — APPROVED** [review]
Looks good to me @gregdhill — a few comments (mostly questions) but no blockers.

File: packages/snap/src/bitcoin/tapSigner.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    checkForInput(psbt.data.inputs, 0); // making sure we have at least one
+    const results = range(psbt.data.inputs.length).map(idx => {
+        const input = psbt.data.inputs[idx];
+        if (isTaprootInput(input)) {
+            return psbt.validateSignaturesOfInput(idx, schnorrValidator);
+        } else {
+            return psbt.validateSignaturesOfInput(idx, validator);
+        }
+    });
+    return results.reduce((final, res) => res === true && final, true);
```
Comment (bvotteler) [inline packages/snap/src/bitcoin/tapSigner.ts:154]: nitpick, not blocking: would be nice to have parentheses for the boolean evaluation here, the evaluation order isn't immediately obvious.

**Review by bvotteler — APPROVED** [review]

File: packages/snap/src/rpc/__tests__/signInput.test.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        signInput: mockSignInput
+      };
+    }),
+    AccountSigner: jest.fn()
+  };
+})
+
+describe('signInput', () => {
+  const snapStub = new SnapMock();
+  const domain = "www.justsnap.io"
```
[the remark this answers is not in the document]
Comment (gregdhill) [reply packages/snap/src/rpc/__tests__/signInput.test.ts:35]: Just tests it is fine

File: packages/snap/src/bitcoin/__tests__/tapSigner.test.ts [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+      ],
+    }))
+    .addOutput({
+      value: 100,
+      address: "1Fh7ajXabJBpZPZw8bjD3QU4CuQ3pRty9u",
+    });
+}
+
+describe('tapSigner', () => {
+  const testRootPrivateKey = "xprv9s21ZrQH143K3GJpoapnV8SFfukcVBSfeCficPSGfubmSFDxo1kuHnLisriDvSnRRuL2Qrg5ggqHKNVpxR86QEC8w35uxmGoggxtQTPvfUu";
```
[the remark this answers is not in the document]
Comment (gregdhill) [reply packages/snap/src/bitcoin/__tests__/tapSigner.test.ts:70]: No should be fine

File: packages/snap/src/bitcoin/__tests__/index.test.ts [hunk 4]
```diff
@@ -85,19 +100,67 @@ changeAddress: tb1qx5wzl3f27d6dgzk9u7h47pqdts60xdpax4w5rf
...
+        })
+        .toBase64();
+
+      const tx = new BtcPsbt(psbtBase64, BitcoinNetwork.Test);
+      const { txId, txHex } = tx.signPsbt(signer);
+      expect(txId).toBe(
+        'e0a70feca465add48d89fade7c66b68589fc95f21c8e69e3121a4bd6cee3302d',
+      );
+      expect(txHex).toBe(
+        '0200000000010100000000000000000000000000000000000000000000000000000000000000000000000000ffffffff016400000000000000160014368c82eb0844679fdd77679bcf193352d57820030140dec9eb88adab2689a69d3e8e32f409c6ed7bb6d7916d1bb780603fd4483849e2dacc27ea1a89c221e66944a84cad617d2489bbb5d69726bfdf3b3390a2a4db4900000000',
```
[the remark this answers is not in the document]
Comment (gregdhill) [reply packages/snap/src/bitcoin/__tests__/index.test.ts:162]: Auto formatting

File: packages/snap/src/bitcoin/__tests__/index.test.ts [hunk 5]
```diff
@@ -85,19 +100,67 @@ changeAddress: tb1qx5wzl3f27d6dgzk9u7h47pqdts60xdpax4w5rf
...
     it('should be able to sign transaction and extract txId and txHex', () => {
-      const tx = new BtcTx(psbtFixture.base64, BitcoinNetwork.Test);
-      const { txId, txHex } = tx.signTx(signer);
+      const tx = new BtcPsbt(psbtFixture.base64, BitcoinNetwork.Test);
+      const { txId, txHex } = tx.signPsbt(signer);
       expect(txId).toBe(
         '4db856b1b7049cce26dc298458796f66d0dee39a5d0651b4c51aa0c66326adec',
       );
       expect(txHex).toBe(
         '020000000001012a4fa3e26d8dbf1c7bcf220e051ef4c74197f374457ccec93b7a6d8445e0ac600000000000ffffffff02400d03000000000016001405b3fe2c8306cc8134b39a5d0512d16ff4f791869acd0b0000000000160014351c2fc52af374d40ac5e7af5f040d5c34f3343d02483045022100cb9909386da8eeda5c505b904a41f3a539d03e9c3af2d2030a6c6bcd3125884802203b922ab5815342501c58dec34317c8a7f02dddccabebbd3deb814dbccb8b4809012103481e8f9077010011be6b251eccc7da40dd22ebe0cee03c31b45fa76ee596a7e200000000',
```
[the remark this answers is not in the document]
Comment (gregdhill) [reply packages/snap/src/bitcoin/__tests__/index.test.ts:114]: Auto formatting

File: packages/snap/src/bitcoin/tapSigner.ts (hunk 1 above)
Comment (gregdhill) [reply packages/snap/src/bitcoin/tapSigner.ts:154]: Copied from bitcoin-js, preserving

**Pull request merged**